general/mm.hpp

```diff
@@ -0,0 +1,136 @@
+// Copyright (c) 2010, Lawrence Livermore National Security, LLC. Produced at
+// the Lawrence Livermore National Laboratory. LLNL-CODE-443211. All Rights
+// reserved. See file COPYRIGHT for details.
+//
+// This file is part of the MFEM library. For more information and source code
+// availability see http://mfem.org.
+//
+// MFEM is free software; you can redistribute it and/or modify it under the
+// terms of the GNU Lesser General Public License (as published by the Free
+// Software Foundation) version 2.1 dated February 1999.
+
+#ifndef MFEM_MM
+#define MFEM_MM
+
+#include <list>
+#include <cstddef> // for size_t
+#include <unordered_map>
+
+using std::size_t;
+
+#include "occa.hpp" // for OccaMemory
+
+namespace mfem
+{
+
+// Implementation of MFEM's lightweight host/device memory manager (mm) designed
+// to work seamlessly with the okina device kernel interface.
+
+/// The memory manager singleton
+class mm
```

Comment:
I'm bothered by these little inconsistencies, why is the class name in lowercase?

Reply:
I would also prefer a more explicit/self documented name, but that might only be my taste.

Reply:
While I don't disagree with both of you, functions like `mm::memcpy` are intended as replacements of `memcpy` so the short and lowercase name makes some sense...

Reply:
I'll put in my vote that even though it requires more typing, I tend to appreciate more descriptive names.